parcels/particle.py

```diff
@@ -251,6 +254,26 @@ def remove(self, indices):
         self.particles = np.delete(self.particles, indices)
         return particles
 
+    def density(self, lon=None, lat=None):
+        if lon is None:
+            lon = self.grid.U.lon
+        if lat is None:
+            lat = self.grid.U.lat
+        Density = np.zeros((lon.size, lat.size), dtype=np.float32)
+        Area = np.zeros(np.shape(Density), dtype=np.float32)
+
+        dx = (lon[1] - lon[0]) * 1852 * 60 * np.cos(lat*math.pi/180)
```

Comment:
Yes! This was written before that PR was finalised. I will familiarise myself with the conversion routines and incorporate.